File: frontend/src/types/generation.ts
export type GenerationRequest = {
  model_id?: number
  prompt_id?: number
  count: number
  target_layer_id: number
}

export type GenerationCandidate = {
  index: number
  statement: string
}

export type GenerationResult = {
  candidates: GenerationCandidate[]
  selected_model_id: number
  selected_model_name: string
  selected_prompt_id: number
  selected_prompt_name: string
}

File: backend/app/schemas/generation.py
"""Generation API schemas."""
from pydantic import BaseModel, Field


class GenerationRequest(BaseModel):
    """Need-to-Spec generation request."""

    model_id: int
    count: int = Field(ge=1, le=10)
    target_layer_id: int | None = None


class GenerationCandidate(BaseModel):
    """Stateless generated candidate."""

    index: int
    statement: str


class GenerationResult(BaseModel):
    """Need-to-Spec generation result."""

    candidates: list[GenerationCandidate]
